### ISSUE TITLE
adding error status codes for invalid queries

### ISSUE THREAD

**Issue by dpastoor** [r-hub/cranlogs.app#41]
Would it be possible to add more appropriate status codes for invalid requests to improve client error handling.

I hit a "bug" while hacking on https://github.com/devOpifex/cranlogs as I'd added some error handling of:

```
if resp.StatusCode != 200 {
		return daily, fmt.Errorf("error getting daily downloads: %s", resp.Status)
	}
```

but then was improperly constructing the string so it was invalid.

Checking via curl shows that there is still a 200 response along with the json `{ "error": "Invalid query",   "info": "https://github.com/metacran/cranlogs.app" }`

```
❯ curl https://cranlogs.r-pkg.org/downloads/dailylast-week/dplyr -v
*   Trying 104.21.58.236:443...
* Connected to cranlogs.r-pkg.org (104.21.58.236) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*  CAfile: /etc/ssl/cert.pem
*  CApath: none
* TLSv1.2 (OUT), TLS handshake, Client hello (1):
* TLSv1.2 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
* TLSv1.2 (IN), TLS handshake, Server finished (14):
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
* TLSv1.2 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.2 (OUT), TLS handshake, Finished (20):
* TLSv1.2 (IN), TLS change cipher, Change cipher spec (1):
* TLSv1.2 (IN), TLS handshake, Finished (20):
* SSL connection using TLSv1.2 / ECDHE-ECDSA-CHACHA20-POLY1305
* ALPN, server accepted to use h2
* Server certificate:
*  subject: C=US; ST=California; L=San Francisco; O=Cloudflare, Inc.; CN=sni.cloudflaressl.com
*  start date: Jul  5 00:00:00 2021 GMT
*  expire date: Jul  4 23:59:59 2022 GMT
*  subjectAltName: host "cranlogs.r-pkg.org" matched cert's "*.r-pkg.org"
*  issuer: C=US; O=Cloudflare, Inc.; CN=Cloudflare Inc ECC CA-3
*  SSL certificate verify ok.
* Using HTTP2, server supports multi-use
* Connection state changed (HTTP/2 confirmed)
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* Using Stream ID: 1 (easy handle 0x148811600)
> GET /downloads/dailylast-week/dplyr HTTP/2
> Host: cranlogs.r-pkg.org
> user-agent: curl/7.77.0
> accept: */*
> 
* Connection state changed (MAX_CONCURRENT_STREAMS == 256)!
< HTTP/2 200 
< date: Fri, 21 Jan 2022 14:48:49 GMT
< x-powered-by: Express
< cf-cache-status: DYNAMIC
< expect-ct: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
< report-to: {"endpoints":[{"url":"https:\/\/a.nel.cloudflare.com\/report\/v3?s=X5Vvidd5ALsPSmETV7yv%2FOMMxheEOOI9Q9cvoeXUmt8pPdTAfwVPf3ocvGaFFFUsH5PsXOdd3TZVyHGZ9%2BQUKPYgxh%2BWCHAJ3nRNSGaMuaiyUrPipE67Jl%2Bqq899FTWJ2pKxQOY%3D"}],"group":"cf-nel","max_age":604800}
< nel: {"success_fraction":0,"report_to":"cf-nel","max_age":604800}
< server: cloudflare
< cf-ray: 6d11587e2e165b4c-IAD
< alt-svc: h3=":443"; ma=86400, h3-29=":443"; ma=86400
< 
* Connection #0 to host cranlogs.r-pkg.org left intact
{ "error": "Invalid query",   "info": "https://github.com/metacran/cranlogs.app" }
```

Ideally would get like a 400, 404 or the like status code

**Comment by dpastoor** [comment]
hmm I think the issue is https://github.com/r-hub/cranlogs.app/blob/master/routes/downloads.js#L125

I'm not sure if that is a valid structure? would t be better to use res.status? https://expressjs.com/en/api.html#res.status

**Comment by dpastoor** [comment]
fixed in #42
